src/global.js

```diff
@@ -35,6 +35,11 @@ export const multiselectControlsList = new Set();
  */
 export const copyData = new Set();
 
+/**
+ * Store the paste shortcut mode.
+ */
+export const inPasteShortcut = new WeakMap();
```

Comment:
I think the primary reason I used a weakmap is because when I paste in workspace 2 after copying from workspace 1, it would highlight one of the blocks from workspace 1. I will probably look into this more when testing out the multiworkspace with the scroll-options plugin (after fixing all the rest of the changes you requested).

Reply:
Ah okay, then ignore the comments here